src/training_hub/algorithms/osft.py

```diff
@@ -58,10 +58,10 @@ def train(
         data_output_dir: str | None = None,
 
         # Torchrun parameters for multi-node support
-        nproc_per_node: int | None = None,
+        nproc_per_node: str | None = None,
```

Comment:
Since you already know the valid string that this could be, I recommend being explicit rather than allowing any string type:

```suggestion
        nproc_per_node: literal["gpu"] | None = None,
```